components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -324,7 +324,11 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
 
         // Redirect to workspaceURL if we are not yet running in an iframe.
         // It happens this late if we were waiting for a docker build.
-        if (!this.props.runsInIFrame && workspaceInstance.ideUrl && !this.props.dontAutostart) {
+        if (
+            !this.props.runsInIFrame &&
+            workspaceInstance.ideUrl &&
+            (!this.props.dontAutostart || workspaceInstance.status.phase === "running")
```

Comment:
:+1:

Reply:
Unfortunately I don't have a reproducer for such a scenario... :thinking:

Reply:
running -> redirect ->   ws-proxy find pod -> ide -> failed -> stopped
                                ｜
                                └>  ws-proxy find no pod -> add not_found -> loop break

@geropl

Reply:
Loop only happens when DB shows workspace status as `running` forever

Reply:
Steps like Anton said https://github.com/gitpod-io/gitpod/pull/9004#issuecomment-1084573937 will repro this scenario, others has been test in prev PR https://github.com/gitpod-io/gitpod/pull/8125

Reply:
> I think after this PR, flow will be like this

@mustard-mh Thx for the flow picture, that explains it! :pray:  We have one failed step, but will break because we still respect `dontAutostart` when triggering the actual workspace start. :heavy_check_mark: :lotus_position: